### PR TITLE
Capi return bool

### LINKED ISSUE #105: CAPI: return boolean instead of custom Success/Failure values
Libvmi has been built with (too) strong compatibility in mind, and used a `VMI_SUCCESS/FAILURE` as a return type when it could have included `stdbool.h` and returned a boolean.

`stdbool.h` only exists since `C99`:
https://stackoverflow.com/a/1608350/3017219

I think we should not try to go lower than this standard.

### PR DESCRIPTION
**Pull request by Wenzel**
fix #105

### CODE REVIEW

File: codecov.yaml [hunk 1]
```diff
@@ -3,3 +3,6 @@ coverage:
     project:
       default:
         informational: true
+    patch:
```
Comment (rageagainsthepc) [inline codecov.yaml:6]: Why is this needed?

File: c_examples/pause.c [hunk 2]
```diff
@@ -1,17 +1,19 @@
+#define _DEFAULT_SOURCE
```
Comment (rageagainsthepc) [inline c_examples/pause.c:1]: What does this do?

File: codecov.yaml (hunk 1 above)
Comment (Wenzel) [reply codecov.yaml:6]: Fix the codecov CI for `codecov/patch` to not fail the CI

File: c_examples/pause.c (hunk 2 above)
Comment (Wenzel) [reply c_examples/pause.c:1]: removes a warning os usleep definition:
https://stackoverflow.com/a/10053817/3017219
